app/app.go

```diff
@@ -556,25 +550,26 @@ func New(
 		appCodec,
 		authtypes.NewModuleAddress(govtypes.ModuleName),
 		keys[feemarkettypes.StoreKey],
-		tkeys[feemarkettypes.TransientKey],
+		tKeys[feemarkettypes.TransientKey],
 		feeSs,
 		app.ConsensusParamsKeeper,
 	)
 	evmSs := app.GetSubspace(evmtypes.ModuleName)
+
 	app.EvmKeeper = evmkeeper.NewKeeper(
 		appCodec,
 		keys[evmtypes.StoreKey],
-		tkeys[evmtypes.TransientKey],
+		tKeys[evmtypes.TransientKey],
 		authtypes.NewModuleAddress(govtypes.ModuleName),
 		app.AccountKeeper,
 		app.BankKeeper,
 		app.StakingKeeper,
 		&app.FeeMarketKeeper,
-		nil,
-		geth.NewEVM,
 		tracer,
 		evmSs,
+		precompiles.StatefulContracts(&app.FungibleKeeper, appCodec, storetypes.TransientGasConfig()),
```

Comment:
Just for my understanding, what is the reason behind using this specific gas config ? 

do you see any value in defining this ourselves ?